src/model/Model_CurrencyHistory.cpp

```diff
@@ -85,41 +85,68 @@ int Model_CurrencyHistory::addUpdate(const int& currencyID, const wxDate& date,
 /** Return the rate for a specific currency in a specific day*/
 double Model_CurrencyHistory::getDayRate(const int& currencyID, const wxString& DateISO)
 {
-    if (currencyID == Model_Currency::GetBaseCurrency()->CURRENCYID)
+    if (currencyID == Model_Currency::GetBaseCurrency()->CURRENCYID || currencyID == -1)
         return 1;
     
     wxDateTime Date;
     Date.ParseDate(DateISO);
     Model_CurrencyHistory::Data_Set Data = Model_CurrencyHistory::instance().find(Model_CurrencyHistory::CURRENCYID(currencyID),Model_CurrencyHistory::CURRDATE(Date));
 
     if (!Data.empty())
+    {
+        //Rate found for specified day
         return Data.back().CURRVALUE;
+    }
+    else if (Model_CurrencyHistory::instance().find(Model_CurrencyHistory::CURRENCYID(currencyID)).size() > 0)
+    {
+        //Rate not found for specified day
+        //Custom query requested to speed-up performances, no way to obtain it with our ORM
+        wxDateTime dFuture, dPast, dNearest;
+
+        const wxString sqlPast = wxString::Format("SELECT MAX(currdate) FROM CURRENCYHISTORY WHERE currencyid = '%i' AND currdate <= '%s';", currencyID, DateISO);
+        wxSQLite3ResultSet rsPast = Model_CurrencyHistory::instance().db_->ExecuteQuery(sqlPast);
+        while (rsPast.NextRow())
+        {
+            dPast.ParseDate(rsPast.GetAsString(0));
+        }
+
+        const wxString sqlFuture = wxString::Format("SELECT MIN(currdate) FROM CURRENCYHISTORY WHERE currencyid = '%i' AND currdate >= '%s';", currencyID, DateISO);
+        wxSQLite3ResultSet rsFuture = Model_CurrencyHistory::instance().db_->ExecuteQuery(sqlFuture);
+        while (rsFuture.NextRow())
+        {
+            dFuture.ParseDate(rsFuture.GetAsString(0));
+        }
+
+        if (dPast.IsValid() && dFuture.IsValid())
+        {
+            const wxTimeSpan spanPast = Date.Subtract(dPast);
+            const wxTimeSpan spanFuture = dFuture.Subtract(Date);
+
+            dNearest = spanPast <= spanFuture ? dPast : dFuture;
+        }
+        else if (dPast.IsValid())
+        {
+            dNearest = dPast;
+        }
+        else if (dFuture.IsValid())
+        {
+            dNearest = dFuture;
+        }
+        else
+        {
+            //TODO: Show warning alert but only one time?
```

Comment:
one time, but per MMEX session. User should be warned every time he runs MMEX that displayed values are wrong because rates are missing.
With buttons: `Download now`, `manually add missing` and `continue [with errors]`

Reply:
OK, I will open separate issue for this (to warn user displayed values are pointless)